controller/linodemachine_controller.go

```diff
@@ -117,17 +160,148 @@ func (r *LinodeMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			LinodeMachine: linodeMachine,
 		},
 	)
+	if err != nil {
+		log.Info("Failed to create machine scope", "error", err.Error())
+
+		return ctrl.Result{}, fmt.Errorf("failed to create machine scope: %w", err)
+	}
 
-	return r.reconcile(ctx, machineScope, clusterScope)
+	return r.reconcile(ctx, machineScope, clusterScope, linodeMachine, log)
 }
 
 func (r *LinodeMachineReconciler) reconcile(
 	ctx context.Context,
 	machineScope *scope.MachineScope,
 	clusterScope *scope.ClusterScope,
-) (ctrl.Result, error) {
-	// TODO
-	return ctrl.Result{}, nil
+	linodeMachine *infrav1.LinodeMachine,
+	logger logr.Logger,
+) (res ctrl.Result, err error) {
+	defer func() {
+		if patchErr := machineScope.PatchHelper.Patch(ctx, linodeMachine); patchErr != nil &&
+			client.IgnoreNotFound(patchErr) != nil {
+
+			logger.Error(patchErr, "failed to patch linodeMachine")
+
+			err = errors.Join(err, patchErr)
+		}
+	}()
+
+	// Delete
+	if !linodeMachine.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.Info("deleting machine")
+
+		if machineScope.LinodeMachine.GetInstanceID() != 0 {
```

Comment:
```suggestion
		if machineScope.LinodeMachine.GetInstanceID() >= 0 {
```

Reply:
maybe just for safety if we get a negative number some how, if you agree to this change then swap out the others you do the same check on